docs/page-objects/fields.rst

```diff
@@ -4,161 +4,312 @@
-    price = await page.price
+    import attrs
+    from web_poet import Returns
+
+    from my_library import BasePage
+
+    @attrs.define
+    class CustomItem:
+        kept_field: str
+
+    @attrs.define
+    class CustomPage(BasePage, Returns[CustomItem], skip_nonitem_fields=True):
+        pass
+
+Alternatively, you can consider :ref:`composition <composition>` for removing
+fields. Composition is more verbose than subclassing, because you need to
+define every field in your page object class, but it can catch some mismatches
+between page object class fields and item class fields that would otherwise be
+hidden by ``skip_nonitem_fields``.
+
+
+.. _rename-field:
+
+Renaming a field
+----------------
+
+To rename a field from a :ref:`page object class <page-objects>` when
+subclassing:
+
+#.  Define a new :ref:`item class <items>` that defines all fields, including
+    the renamed field.
+
+#.  In your new page object class, subclass the original page object class,
+    :class:`~.Returns` with the new item class between brackets, and set
+    ``skip_nonitem_fields=True``.
 
-This is not ideal, because now the code which needs to use a page object
-must be aware if a field is sync or async. If a field needs to be changed
-from being sync to ``async def`` (or the other way around),
-e.g. because of a website change, all the code which uses this page
-object must be updated.
+    When building an item, page object class fields without a matching item
+    class field will now be ignored, rather than raising an exception.
 
-One approach to solve it is to always define all fields as ``async def``.
-It works, but it makes the page objects harder to use in non-async environments.
+#.  Define a field for the new field name that returns the value from the old
+    field name.
 
-Instead of doing this, you can also use :func:`~.ensure_awaitable` utility
-function when accessing the fields:
+Your new page object class will still define the old field name, but the
+resulting item will not.
+
+For example:
 
 .. code-block:: python
 
-    from web_poet.utils import ensure_awaitable
+    import attrs
+    from web_poet import Returns
 
-    page = MyPage(...)
-    name = await ensure_awaitable(page.name)
-    price = await ensure_awaitable(page.price)
+    from my_library import BasePage
+
+    @attrs.define
+    class CustomItem:
+        new_field: str
+
+    @attrs.define
+    class CustomPage(BasePage, Returns[CustomItem], skip_nonitem_fields=True):
+
+        @field
+        async def new_field(self):
+            return ensure_awaitable(self.old_field)
+
+Alternatively, you can consider :ref:`composition <composition>` for renaming
+fields. Composition is more verbose than subclassing, because you need to
+define every field in your page object class, but it can catch some mismatches
+between page object class fields and item class fields that would otherwise be
+hidden by ``skip_nonitem_fields``.
+
+
+.. _composition:
+
+Composition
```

Comment:
Nice! 💯 

Shall we also cover the case when the PO uses an item directly? For example: https://github.com/scrapinghub/scrapy-poet/blob/master/tests/test_web_poet_rules.py#L1054-L1065

Reply:
What do you think about [this](https://github.com/scrapinghub/web-poet/pull/193/commits/41d6c743a070f804968750c9efdc26adb8afc032)?